### PR TITLE
Run rustfmt with default settings. What looks bad?

### PR DESCRIPTION
**Pull request by callahad**
Some of the formatting choices aren't great for us, but this is our starting point.

...now to figure out which ones we care about, and how to configure rustfmt to work better for us.

**NOTE:** This is  a work-in-progress discussion; not necessarily planning to merge this branch directly!

### CODE REVIEW

**Comment by callahad** [comment]
Odd. Seems like rustfmt takes a few invocations before it settles?

**Comment by callahad** [comment]
Weird! It'll work just find for a few invocations, and then it'll reformat something.

If I run it enough times, it'll format it back:

screenshot from 2016-10-12 21-19-17


So.... seems like `cargo fmt` isn't deterministic, yet. Upstream bug, here I come.

**Comment by callahad** [comment]
For instance, consider these three consecutive runs:

e9970cbb37d7aafc64801ff887bb0203ee505fda indents some comments
e3d709505d9fd5017a601de53074870fc1537d80 de-indents the same comments
345fc2825f1bc22ca29f1cbe4f77fc7da89be238 re-indents the same comments

WTF?

File: src/store_cache.rs [hunk 1]
```diff
@@ -73,19 +70,21 @@ impl StoreCache {
 
             Ok(data)
 
-        }, |data| {
-            Ok(data)
-        })
+        },
+                           |data| Ok(data))
```
Comment (stephank) [inline src/store_cache.rs]: This one is especially weird. Looks like it tried to align with the `(` from `map_or_else`. Though `map_or_else` felt weird to use anyway, so I wouldn't mind if this was rewritten.

**Comment by callahad** [comment]
Closing this for now - we'll revisit rustfmt post 1.0 :)

**Pull request closed without merging**